### DISCUSSION TITLE
Saturday Update

### DISCUSSION THREAD

**Discussion by MikasaSucasa** [frc5567/2023-Robot#207] [Team Posts]
All, I want to say thank you for all the hard work we have done thus far. The robot is running 2021 software and firmware and still moves. This is great news since it was our first time seeing the robot. There is still a ton to do. We are still reviewing the proto pull request but when that is approved, we need to keep making changes.
1) we should ignore the PID for launcher and just set the speed to 0.9 or something
2) we should include a slow feature that allows the driver to move in small increments for launching (half the speed or something).
3) auton needs some serious debugging but I think we need to add the integer values back in. Coders that were there should do some research into this. Please feel free to update the team on the exact issue that was occurring.

We will most likely be back ASAP to resolve the launching and basic auton issues

**Comment by JooshO** [comment]
Can you (or anyone who was there) elaborate on why you think not having the integer values would be causing an issue (I assume this is in reference to the auton enum)? Seeing as we never reference them and they aren't required for testing if values are equal, I don't see how they could cause an issue

**Comment by MikasaSucasa** [comment]
It isn't so much as resolving the issue but having a to string or value might help with debugging was my thought. The issue that I was hoping someone would describe is that after assignment the step1 value is never reached but it is always step2. I have not looked at the default toString method. Perhaps it will give index as the compareTo method knows relative position (so I know it is built into the object somehow). We do weird things such as define it in constructor as well as init but I didn't see quick solution. Perhaps the conditional is wrong in the way that we reference the enum?

**Comment by JooshO** [comment]
~~Well I think at least part of the reason step 2 was never reached would be that the turnToAngle will never exit like was mentioned in a comment on the pull request.~~ (this would be if it was stuck at step 2)

Was step2 never reached or did it never exit step2?

**Comment by MikasaSucasa** [comment]
no step1 was never reached...

**Comment by MikasaSucasa** [comment]
yea, skipped the init stage

**Comment by JooshO** [comment]
So it went straight to step 2 and skipped step 1? The robot never drove forward? Or did it not go at all

This could be an issue caused by auton init not running every time if we're just re-running auton although I doubt that would be it

**Comment by HannahStrimpel** [comment]
The robot didn't go forward. It just spun clockwise.

**Comment by JooshO** [comment]
And I assume this behaviour persisted between pushes/rebuilds/power cycles?

**Comment by HannahStrimpel** [comment]
yes

**Comment by JooshO** [comment]
Well something to note that could be causing an early exit, although I doubt that it is related directly to this, is that zeroEncoders resets the SensorCollections that we create to represent our encoders, but when we call getLeftEncoderPostion we use .getSelectedSensorPosition on the talon itself.

As far as I can see, we never configure the selected sensor for those motor controllers, so this conflict could be causing unexpected or undefined behavior. Would definitely be worthwhile to compare outputs from selectedSensorPosition and the SensorCollection getQuadraturePosition

Edit - Methods to look at in the ProtoBranch:
zeroEncoders(), getLeftDriveEncoderPosition, and the constructor in Drivetrain
configSelectededFeedbackSensor() on the TalonFX

ex of configSelectedFeedbackSensor from last year's code:
![image](https://user-images.githubusercontent.com/25639881/111400959-c26fe580-869e-11eb-85b3-7f43d0b92d79.png)

**Comment by JooshO** [comment]
Basically, there is potential that the encoders that we read position from are never actually reset
